firmware/boron-ins-fsm/test/mocks/mock_eeprom.h

```diff
@@ -1,21 +1,45 @@
+/*
+ * Mock EEPROM memory for Particle Boron
+ * Reference: https://docs.particle.io/reference/device-os/api/eeprom/eeprom/
+ * 
+ * NOTE: Due to an issue where sizeof() returns different values on different 
+ * operating systems for certain data types, this mock EEPROM implementation
+ * differs from the actual Boron EEPROM by always allocating BYTES_PER_ELEMENT
+ * bytes for each element being stored. In reality, the EEPROM only has 
+ * BORON_EEPROM_SIZE bytes of memory. When data is read from or written to the
+ * address pointed to by _address, sizeof(data) is used to read or write a
+ * number of bytes equal to the size of the data type. 
+*/
+
 #pragma once
 
+#define BORON_EEPROM_SIZE 4096
+#define BYTES_PER_ELEMENT 8
+
 class MockEEPROM
 {
-public:
-    MockEEPROM()
-    {
-    }
+private:
+    uint8_t mockMemory[BORON_EEPROM_SIZE * BYTES_PER_ELEMENT];
 
 public:
+    MockEEPROM() {}
+
     template <typename T>
-    void get(int const _address, T& data)
-    {
+    void get(int const _address, T& data) {
+        if (_address >= BORON_EEPROM_SIZE) {
+            return;
+        }
+
+        memcpy(&data, mockMemory + (_address * BYTES_PER_ELEMENT), sizeof(data));
     }
 
     template <typename T>
-    void put(int const _address, T const& _data)
-    {
+    void put(int const _address, T const& _data) {
+        if (_address >= BORON_EEPROM_SIZE) {
+            return;
+        }
+
+        memcpy(mockMemory + (_address * BYTES_PER_ELEMENT), &_data, sizeof(_data));
```

Comment:
I still think that this is too complicated for what we want to do here. Just the fact that you ran into issues with the size of memory shows that it is fragile and that the test can fail for a reason outside the code that it's supposed to be testing.

I would prefer that the `get` be implemented to always return fixed values. It can be different for each address (if that would make the test better) by putting a if/else inside that will return different values depending on what `_address` is.

And the PUT doesn't have to do anything. I think you'll still be able to do your two tests even with this simplified mock because `e` will return a value that you can use and the positive int is processed in variables to form the return value, it isn't re-retrieved from EEPROM. What do you think?

Reply:
This works for me. I've made the changes in my latest commit.